### PR TITLE
make sure the email is valid , otherwise return error

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:

 make sure the email is valid.

**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35955)

### CODE REVIEW

**Comment by ericchiang** [comment]
Please take a look at https://github.com/kubernetes/kubernetes/pull/36087 as well.

Kubernetes already trusts the OpenID Connect provider to a high degree, since it will respect any id token that that server signs (e.g. even a malicious one if the provider so chooses).

Basically, I'm not sure what we gain by this check. Why have this in Kubernetes? What providers return invalid emails? Shouldn't it be the responsibility of the provider?

File: plugin/pkg/auth/authenticator/token/oidc/oidc.go [hunk 1]
```diff
@@ -44,6 +45,13 @@ import (
...
...|\\d|[!#\\$%&'\\*\\+\\-\\/=\\?\\^_`{\\|}~]|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])+(\\.([a-zA-Z]|\\d|[!#\\$%&'\\*\\+\\-\\/=\\?\\^_`{\\|}~]|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])+)*)|((\\x22)((((\\x20|\\x09)*(\\x0d\\x0a))?(\\x20|\\x09)+)?(([\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x7f]|\\x21|[\\x23-\\x5b]|[\\x5d-\\x7e]|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])|(\\([\\x01-\\x09\\x0b\\x0c\\x0d-\\x7f]|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}]))))*(((\\x20|\\x09)*(\\x0d\\x0a))?(\\x20|\\x09)+)?(\\x22)))@((([a-zA-Z]|\\d|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])|(([a-zA-Z]|\\d|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])([a-zA-Z]|\\d|-|\\.|_|~|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])*([a-zA-Z]|\\d|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])))\\.)+(([a-zA-Z]|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])|(([a-zA-Z]|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])([a-zA-Z]|\\d|-|\\.|_|~|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])*([a-zA-Z]|[\\x{00A0}-\\x{D7FF}\\x{F900}-\\x{FDCF}\\x{FDF0}-\\x{FFEF}])))\\.?$"
```
Comment (ericchiang) [inline plugin/pkg/auth/authenticator/token/oidc/oidc.go]: where is this coming from?

Comment (tanshanshan) [inline plugin/pkg/auth/authenticator/token/oidc/oidc.go]: I get the regexp from 
> https://github.com/asaskevich/govalidator/blob/master/patterns.go
     and the validator method in file
> https://github.com/asaskevich/govalidator/blob/master/validator.go
the method name is   **IsEmail**

**Comment by liggitt** [comment]
agree, I don't think email syntax validation of the claim returned from the IDP adds much value

**Pull request closed without merging**